src/main/java/org/jabref/gui/preferences/TableColumnsTab.java

```diff
@@ -141,36 +137,35 @@ public TableColumnsTab(JabRefPreferences prefs, JabRefFrame frame) {
         HBox toolBar = new HBox();
         Button addRow = new Button("Add");
         addRow.setPrefSize(80, 20);
-        addRow.setOnAction( e -> {
-            if (!addLast.getText().isEmpty()) {
-                TableRow tableRow = addLast.getText().matches("[1-9][0-9]") ? new TableRow(addName.getText(), Integer.valueOf(addLast.getText())) : new TableRow(addName.getText());
-                addName.clear();
-                addLast.clear();
-                data.add(tableRow);
-                tableRows.clear();
-                tableRows.addAll(data);
-                colSetup.setItems(data);
-                tableChanged = true;
-                colSetup.refresh();
-            }
+        addRow.setOnAction(e -> {
+            TableRow tableRow = new TableRow(addName.getText());
+            addName.clear();
+            data.add(tableRow);
+            tableRows.clear();
```

Comment:
Shouldn't it be sufficient to add the TableRow only to the data. The TavleView should update automatically.